shared/api/instance_backup.go

```diff
@@ -2,28 +2,44 @@ package api
 
 import "time"
 
-// ContainerBackupsPost represents the fields available for a new LXD container backup
-// API extension: container_backup
-type ContainerBackupsPost struct {
+// InstanceBackupsPost represents the fields available for a new LXD instance backup.
+//
+// API extension: instances
+type InstanceBackupsPost struct {
 	Name             string    `json:"name" yaml:"name"`
 	ExpiresAt        time.Time `json:"expires_at" yaml:"expires_at"`
 	ContainerOnly    bool      `json:"container_only" yaml:"container_only"`
```

Comment:
Can you prepare a separate branch which replaces all instances of `ContainerOnly` by `Snapshots` (so reversing the logic) but keeping the same defaults.

Logic should then be that:
 - If ContainerOnly is set to true => Set Snapshots to false
 - If ContainerOnly is set to false => Set Snapshots to true
 - If ContainerOnly isn't set => Depend on Snapshots only
 - If Snapshots isn't set => true
 - If Snapshots and !ContainerOnly don't line up, error

On the server side, when unpacking the received JSON, you can set defaults in the struct that you're about to fill, so something like this:
```
req := api.InstanceBackupsPost{}
req.ContainerOnly = False
req.Snapshots = True
yaml.Unmarshal(input, &req)
```

We have this present in both the backup and migration API, so both of those would need updating the same way.